services/validators.go

```diff
@@ -48,6 +49,27 @@ func ValidateAdapter(bt *models.BridgeType, store *store.Store) (err error) {
 	return fe.CoerceEmptyToNil()
 }
 
+// ValidateMinimumContractPayment returns true when the amount is >= the cost of
+// every adapter used in the job
+func ValidateMinimumContractPayment(
+	s *store.Store,
+	job models.JobSpec,
+	amount assets.Link,
+) (bool, error) {
+	cost := assets.NewLink(0)
+
+	for _, task := range job.Tasks {
+		a, err := adapters.For(task, s)
+		if err != nil {
+			return false, fmt.Errorf("policies.MinimumContractPaymentValid: %v", err)
+		}
+		mp := a.MinContractPayment()
+		cost.Add(cost, &mp)
+	}
+
+	return cost.Cmp(&amount) <= 0, nil
```

Comment:
I'm of the opinion that `ValidateMinimumContractPayment` should return an error if the amount is < cost of every adapter. This might make the first return value (bool) redundant.

That might allow you to DRY up the section @j16r is referring to: https://github.com/smartcontractkit/chainlink/pull/623/files/529f36483d24205385052ff6973915f13d94408d#r223064288

Reply:
@dimroc interesting. I pulled out the common code as a new commit but I do see that the other `Validate...` functions return an error so I will use that pattern.